haskell-src/Concordium/MerkleProofs.hs

```diff
@@ -596,7 +597,10 @@ blockSchema = runState builder emptySchemaBuilderState & _2 %~ _builderSchema
                   Hashed "result" resultHash
                 ]
         blockQuasiHash <- setFresh . Sequence $ [Hashed "meta" metaHash, Hashed "data" dataHash]
-        schemaAt blockHash ?= Sequence [Hashed "header" blockHeaderHash, Hashed "quasi" blockQuasiHash]
+        rootHash <-
+            setFresh . Sequence $
+                [Hashed "header" blockHeaderHash, Hashed "quasi" blockQuasiHash]
+        schemaAt blockHash ?= Sequence [LiteralBytes (S.encode P7), Hashed "root" rootHash]
```

Comment:
Maybe `blockSchema` should be named such that it reflects that it's only useful for P7? Could the function be parameterized by some "BlockHashVersion" that is determined from the protocol version?

Reply:
I think in future the schema could/should support parsing for multiple protocol versions. At the moment, it's only well-defined for protocol version 7.